src/panels/alarm-configuration/AlarmEditModal.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback, useState } from 'react';
+import { Button, Field, Input, Modal } from '@grafana/ui';
+
+interface AlarmEditProps {
+  isOpen: boolean;
+  currentThreshold: number;
+  onDismiss: () => void;
+  onSave: (newThreshold: number) => void;
+}
+
+export const AlarmEditModal: React.FunctionComponent<AlarmEditProps> = ({
+  isOpen,
+  onDismiss,
+  onSave,
+  currentThreshold,
```

Comment:
This is good, but I'd see about getting these from context, rather than props, or add a decorator component that connects this to state.